File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        a=[]
        while head!=None:
            a.append(head.val)
            head=head.next
        del a[-n]
        dummy = ListNode(0)
        temp = dummy
        for val in a:
            temp.next = ListNode(val)
            temp = temp.next
        return dummy.next
        
        

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        int l=0;
        ListNode*ptr=head;
        while(ptr!=nullptr){
            l++;
            ptr=ptr->next;
        } 
        if (n == l) {
            ListNode* temp = head;
            head = head->next;
            delete temp;
            return head;
        }
        int r=l-n;
        ListNode* ptr1=head;
        int c=1;
        while(c<r){
            ptr1=ptr1->next;
            c++;
        }
        ListNode* temp=ptr1->next;
        ptr1->next=ptr1->next->next;
        delete temp;
        return head;
    }
};